src/test/rgw/rgw_multi/tests.py

```diff
@@ -957,9 +963,22 @@ def test_datalog_autotrim():
 
     # trim each datalog
     for zone, _ in zone_bucket:
+        # read max markers for each shard
+        status = datalog_status(zone.zone)
+
         datalog_autotrim(zone.zone)
-        datalog = datalog_list(zone.zone)
-        assert len(datalog) == 0
+
+        for shard_id, shard_status in enumerate(status):
+            try:
+                before_trim = dateutil.parser.isoparse(shard_status['last_update'])
+            except: # empty timestamps look like "0.000000" and will fail here
+                continue
+            entries = datalog_list(zone.zone, ['--shard-id', str(shard_id), '--max-entries', '1'])
+            try:
+                after_trim = dateutil.parser.isoparse(entries[0]['timestamp'])
```

Comment:
could there be a case where you get zero entries back?

Reply:
yes, in that case `entries[0]` raises an `IndexError` that gets ignored